floc.bs

```diff
@@ -1 +1,149 @@
+</section>
+
+<section>
+  <h2 id="interpretation-from-ad-techs">Interpretation from ad techs</h2>
+  Ad tech companies are supposed to observe the habits of each <a href="#string-representation">interest cohort</a> and ad targeting can then be partly based on what group the person falls into. In general, different <a href="#version">versions</a> should not be compatible. However, the browser vendors could share additional context of the version and/or could provide recommendations on the compatibility between versions.
+</section>
+
+<section>
+  <h2 id="algorithm">Algorithm</h2>
+  <em>This section uses "cohort" as the short name for "<a>interest cohort id</a>".</em>
+
+  <h3 id="cohort-assignment-algorithm">Cohort assignment algorithm</h3>
+  The browser could use machine learning algorithms to develop the cohort.
+
+  <h4 id="input-features">Input features</h4>
+  The input features to the algorithm should be based on information from the visited sites, which may include the URLs, the page contents, or other factors. The input features should be kept local on the browser and should not be uploaded elsewhere — the browser only exposes the generated cohort.
+
+  <h4 id="anonymity-and-privacy-guarantees">Anonymity and Privacy guarantees</h4>
+  The browser should ensure that cohorts are well distributed, so that each represents thousands of people. The browser may further leverage other anonymization methods, such as differential privacy.
+
+  The browser should ensure that the interest cohorts are not correlated with <a href="#sensitive-information">sensitive information</a>.
+
+  The browser may give the user an empty cohort to meet the above requirements.
+
+  <h3 id="updating-invalidating-the-cohort">Updating/Invalidating the cohort</h3>
+  The browser should recompute the cohort on a regular basis so that the user’s latest browsing behavior can be captured. The browser should also limit the computation frequency to mitigate the <a href="#tracking-people-via-their-interest-cohort">fingerprinting risks</a>.
+
+  The browser should respect any change in user preferences (i.e. dedicated permission setting, or cookie setting at the <a href="#adoption-phase">adoption phase</a>) and may need to clear the interest cohort accordingly.
+
+  When the browsing history is deleted and the current interest cohort is derived from any deleted browsing history, the interest cohort should be cleared.
+</section>
+
+<section>
+  <h2 id="privacy-considerations">Privacy Considerations</h2>
+
+  <h3 id="permission">Permission</h3>
+  <h4 id="compute-eligibility">Eligibility for a page to be included in the interest cohort computation</h4>
+  By default, a page is eligible for the interest cohort computation if the <a href="#the-api">document.interestCohort</a> API was used in the page during the page load.
+
+  The page can opt-out itself for the interest cohort computation through the interest-cohort <a>permissions policy</a> header.
```

Comment:
This permissions policy needs to be defined somewhere. It's pretty simple; see examples like https://html.spec.whatwg.org/multipage/infrastructure.html#policy-controlled-features and https://xhr.spec.whatwg.org/#feature-policy-integration

Reply:
Done.